app/src/main/java/zapsolutions/zap/channelManagement/ChannelItemAdapter.java

```diff
@@ -14,18 +15,54 @@
 
 public class ChannelItemAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
 
-    private List<ChannelListItem> mItems;
+    private final SortedList<ChannelListItem> mSortedList = new SortedList<>(ChannelListItem.class, new SortedList.Callback<ChannelListItem>() {
+        @Override
+        public int compare(ChannelListItem i1, ChannelListItem i2) {
+            return i2.compareTo(i1);
+        }
+
+        @Override
+        public void onChanged(int position, int count) {
+            notifyItemRangeChanged(position, count);
+        }
+
+        @Override
+        public boolean areContentsTheSame(ChannelListItem oldItem, ChannelListItem newItem) {
+            return oldItem.equals(newItem);
+        }
+
+        @Override
+        public boolean areItemsTheSame(ChannelListItem item1, ChannelListItem item2) {
+            return item1.isSameChannel(item2);
```

Comment:
Maybe I understood it wrong, but isn't that exactly what I am doing?
I check if the unique item id is equal. But what is the unique item id? I have chosen the ChannelID for open channels and the ChannelPoint for pending channels as the "UniqueID".
The equals method checks the ChannelByteString right now which should change if there was a new transaction for example.
This means if we would fetch a channel update and a channel had a new transaction, the equal method would return false.
If we then add the channel to our list we would have the channel twice in the list, as it is recognized as a new channel.
If we use the isSameChannel() method, it would recognize it as same channel and just update the entry in the list.

Now all of this isn't actually necessary as for now we always replace all of the entries when we fetch a new channel update.
But this way we could later implement a listener that listens to routed lightning payments for example and could display the changes in real time.